larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -71,23 +80,17 @@ void* Init(TfLiteContext* context, const char* buffer, std::size_t length) {
   }
 
   // reading the op's input arguments into the "conv_params" struct
-  // readng strides
-  auto strides_vector = m["strides"].AsTypedVector();
-  if (strides_vector.size() != 4) {
-    context->ReportError(context, "Strides vector should have size 4.");
-    return conv_params;
-  }
-  for (std::size_t i = 0; i < strides_vector.size(); ++i)
-    conv_params->strides[i] = strides_vector[i].AsInt64();
-
+  LCE_ENSURE_PARAM(conv_params, context, !m["stride_height"].IsNull());
+  LCE_ENSURE_PARAM(conv_params, context, !m["stride_width"].IsNull());
+  LCE_ENSURE_PARAM(conv_params, context, !m["dilation_height_factor"].IsNull());
+  LCE_ENSURE_PARAM(conv_params, context, !m["dilation_width_factor"].IsNull());
+
+  // reading strides
+  conv_params->stride_height = m["stride_height"].AsInt64();
+  conv_params->stride_width = m["stride_width"].AsInt64();
   // reading dilations
-  auto dilation_vector = m["dilations"].AsTypedVector();
-  if (dilation_vector.size() != 4) {
-    context->ReportError(context, "Dilations vector should have size 4.");
-    return conv_params;
-  }
-  for (std::size_t i = 0; i < dilation_vector.size(); ++i)
-    conv_params->dilations[i] = dilation_vector[i].AsInt64();
+  conv_params->dilation_height_factor = m["dilation_height_factor"].AsInt64();
+  conv_params->dilation_width_factor = m["dilation_width_factor"].AsInt64();
 
   // reading padding
   if (m["padding"].ToString() == "VALID" ||
```

Comment:
I'm not sure why we originally had both the upper and lowercase checks here. I think we can just remove one of them? I know its not entirely relevant to this PR, but since we're updating the attributes I think we can do this here as well.

Reply:
I can check this out in a future PR and see if it is cleaner to directly pass the TFLite padding enum (same can be done for the activation strings then as well).